packages/strapi-plugin-graphql/services/Loaders.js

```diff
@@ -0,0 +1,187 @@
+'use strict';
+
+/**
+ * Loaders.js service
+ *
+ * @description: A set of functions similar to controller's actions to avoid code duplication.
+ */
+
+const _ = require('lodash');
+const DataLoader = require('dataloader');
+
+module.exports = {
+  loaders: {},
+
+  initializeLoader: function() {
+    // Create loaders for each relational field (exclude core models).
+    Object.keys(strapi.models)
+      .filter(model => model !== 'core_store')
+      .forEach(model => {
+        (strapi.models[model].associations || []).forEach(association => this.createLoader(association.collection || association.model, association.plugin));
+      });
+
+    // Reproduce the same pattern for each plugin.
+    Object.keys(strapi.plugins).forEach(plugin => {
+      Object.keys(strapi.plugins[plugin].models).forEach(model => {
+        (strapi.plugins[plugin].models[model].associations || []).forEach(association => this.createLoader(association.collection || association.model, association.plugin));
```

Comment:
You're probably creating more than once the loader, why don't you create a loader for the model only and not the association models, your code could be simplified:
```JS
_.chain(strapi.plugins)
    .map('models')
    .flatten()
    .concat(strapi.models)
    .reject((v, k) => k === 'core_store')
 // .map() You can build the model name here or you can just access the collectionName directly
    .forEach(this.createLoader);
```

Reply:
The code loops on each plugin and each plugin's models. There is no duplicate. The difference is that my code seems to be more maintainable than yours. I was developing with a lot of Lodash functions like you did before but it can be very difficult to understand what it does and it's less maintainable for the others developers who are going to add new lines of code.

I'll keep this code like this until you can prove me that there are duplicated loaders 🙈 

Thank you for the feedback, I really appreciate 👍

Reply:
I don't 100% agree with you and we can verify which of our versions is more maintainable / comprehensive (This is not the main question though). The main problem with your version is that you're looping through the associations of each model / plugin's model which mean that if you have Model A and B that have an association to the model C you'll create two times the loader for model C (I'm probably missing something 🙈 )

Reply:
Oh yes! You're right. I'll add a `resetLoaders` method then I'll add a condition to return the loader if it's already existing instead of creating a new one.